apps/glusterfs/operations_block_volume.go

```diff
@@ -376,6 +376,295 @@ func (bvc *BlockVolumeCreateOperation) CleanDone() error {
+			return err
+		}
+
+		bve.op.RecordExpandBlockVolume(bv, bve.newSize)
+		if e := bve.op.Save(tx); e != nil {
+			return e
+		}
+		return nil
+	})
+}
+
+func (bve *BlockVolumeExpandOperation) Exec(executor executors.Executor) error {
+	var (
+		err     error
+		bv      *BlockVolumeEntry
+		hvname  string
+		bvHosts nodeHosts
+	)
+	err = bve.db.View(func(tx *bolt.Tx) error {
+		txdb := wdb.WrapTx(tx)
+		bv, err = NewBlockVolumeEntryFromId(tx, bve.bvolId)
+		if err != nil {
+			return err
+		}
+		hvname, err = bv.blockHostingVolumeName(txdb)
+		if err != nil {
+			return err
+		}
+		bvHosts, err = bv.hosts(txdb)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		logger.LogError(
+			"failed to get state needed to expand block volume: %v", err)
+		return err
+	}
+	// nothing past this point needs a db reference
+	logger.Info("executing expand of block volume %v in op:%v",
+		bve.bvolId, bve.op.Id)
+	return newTryOnHosts(bvHosts).once().run(func(h string) error {
+		err := executor.BlockVolumeExpand(h, hvname, bv.Info.Name, bve.newSize)
+		if err != nil {
+			logger.LogError("Unable to Expand volume: %v", err)
+			return err
+		}
+		return nil
+	})
+}
+
+func (bve *BlockVolumeExpandOperation) Rollback(executor executors.Executor) error {
+	logger.Info("Starting Rollback for %v op:%v", bve.Label(), bve.op.Id)
+	return rollbackViaClean(bve, executor)
+}
+
+func (bve *BlockVolumeExpandOperation) Finalize() error {
+	return bve.db.Update(func(tx *bolt.Tx) error {
+		var bv *BlockVolumeEntry
+		bv, e := NewBlockVolumeEntryFromId(tx, bve.bvolId)
+		if e != nil {
+			return e
+		}
+
+		bv.Info.Size = bve.newSize
+
+		bve.op.FinalizeBlockVolume(bv)
+		if e := bv.Save(tx); e != nil {
+			return e
+		}
+
+		bve.op.Delete(tx)
+		return nil
+	})
+}
+
+func (bve *BlockVolumeExpandOperation) Clean(executor executors.Executor) error {
+	logger.Info("Starting Clean for %v op:%v", bve.Label(), bve.op.Id)
+	var (
+		err error
+		bv  *BlockVolumeEntry
+		// hvname is the name of the hosting volume and is required
+		hvname string
+
+		// host mappings for exec-on-up-host try util
+		bvHosts nodeHosts
+
+		actualSize int
+	)
+	err = bve.db.View(func(tx *bolt.Tx) error {
+		txdb := wdb.WrapTx(tx)
+		bv, err = NewBlockVolumeEntryFromId(tx, bve.bvolId)
+		if err != nil {
+			return err
+		}
+		hvname, err = bv.blockHostingVolumeName(txdb)
+		if err != nil {
+			return err
+		}
+		bvHosts, err = bv.hosts(txdb)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		logger.LogError(
+			"failed to get state needed to get info of block volume: %v", err)
+		return err
+	}
+	// nothing past this point needs a db reference
+	logger.Info("executing get info of block volume %v in op:%v",
+		bve.bvolId, bve.op.Id)
+	err = newTryOnHosts(bvHosts).once().run(func(h string) error {
+		bvolInfo, err := bv.BlockVolumeInfoFromHost(executor, hvname, h)
+		if err != nil {
+			return err
+		}
+		actualSize = bvolInfo.Size
```

Comment:
Unless you have a reason not to you could just assign to `bve.Reclaim.actualSize` here and skip the intermediate variable. But it's not a big deal.